classes/decision/types/BackToPreviousInternalReviewRound.inc.php

```diff
@@ -0,0 +1,228 @@
+
+        $request = Application::get()->getRequest();
+
+        $reviewRoundDao = DAORegistry::getDAO('ReviewRoundDAO'); /** @var ReviewRoundDAO $reviewRoundDao */
+        $reviewAssignmentDao = DAORegistry::getDAO('ReviewAssignmentDAO'); /** @var ReviewAssignmentDAO $reviewAssignmentDao */
+        $reviewRoundId = (int)$request->getUserVar('reviewRoundId');
+
+        $reviewAssignmentDao->deleteByReviewRoundId($reviewRoundId);
+        $reviewRoundDao->deleteById($reviewRoundId);
+    }
+
+    public function getSteps(Submission $submission, Context $context, User $editor, ?ReviewRound $reviewRound): ?Steps
+    {
+        $steps = new Steps($this, $submission, $context, $reviewRound);
+
+        $fakeDecision = $this->getFakeDecision($submission, $editor);
+        $fileAttachers = $this->getFileAttachers($submission, $context);
+
+        $authors = $steps->getStageParticipants(Role::ROLE_ID_AUTHOR);
+        if (count($authors)) {
+            $mailable = new DecisionBackToPreviousInternalReviewRoundNotifyAuthor($context, $submission, $fakeDecision);
+            $steps->addStep(new Email(
+                $this->ACTION_NOTIFY_AUTHORS,
+                __('editor.submission.decision.notifyAuthors'),
+                __('editor.submission.decision.backToPreviousInternalReviewRound.notifyAuthorsDescription'),
+                $authors,
+                $mailable
+                    ->sender($editor)
+                    ->recipients($authors),
+                $context->getSupportedFormLocales(),
+                $fileAttachers
+            ));
+        }
+
+        $reviewAssignments = $this->getActiveReviewAssignments($submission->getId(), $reviewRound->getId());
+        if (count($reviewAssignments)) {
+            $reviewers = $steps->getReviewersFromAssignments($reviewAssignments);
+            $mailable = new DecisionBackToPreviousInternalReviewRoundNotifyReviewer($context, $submission, $fakeDecision);
+            $steps->addStep((new Email(
+                $this->ACTION_NOTIFY_REVIEWERS,
+                __('editor.submission.decision.notifyReviewers'),
+                __('editor.submission.decision.backToPreviousInternalReviewRound.notifyReviewers.description'),
+                $reviewers,
+                $mailable->sender($editor),
+                $context->getSupportedFormLocales(),
+                $fileAttachers
+            ))->canChangeRecipients(true));
+        }
+
+        return $steps;
+    }
+
+    /**
+     * Determine if it is possible to back out form current internal review round to previous internal review round
+     */
+    public function canRetract(Submission $submission, ?int $reviewRoundId): bool
+    {
+        if (!$reviewRoundId) {
+            return false;
+        }
+
+        // if no reviewer assigned , then back out is not proper choice
+        // but to remove the round is the right choice
+        if (!$this->hasReviewerAssigned($reviewRoundId)) {
+            return false;
+        }
```

Comment:
This is done as there is also 2 more decisions as 
1. RemoveEmptyExternalReviewRound
2. RemoveEmptyInternalReviewRound

so this is done so that when there is no reviewer assigned , option would be remove but to back out . But now I am thinking that rather than just have it say `Cancel Review Round` and remove the both of `RemoveEmptyXXXXXXXXReviewRound`. that way we can have less code and decision to tackle .

Reply:
:+1:

Reply:
Done .

Reply:
Done .

Reply:
Done .